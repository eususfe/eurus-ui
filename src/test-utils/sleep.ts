/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
