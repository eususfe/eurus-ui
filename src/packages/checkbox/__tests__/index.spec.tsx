import { mount } from '@vue/test-utils';
import { h } from 'vue';
import { expect } from 'vitest';
import ECheckbox from '..';
import ECheckboxGroup from '../src/checkbox-group';

describe('ECheckbox', () => {

  it('ECheckbox snapshot', () => {
    const wrapper = mount(ECheckbox);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should emit change event', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckbox, {
      props: {
        modelValue: false,
        onChange: fn

      }
    }
    );
    await wrapper.find('input').trigger('change');
    expect(fn).toHaveBeenCalled();
  });

  it('shod not emit change event when disabled', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckbox, {
      props: {
        modelValue: false,
        onChange: fn,
        disabled: true
      }
    });
    await wrapper.find('input').trigger('change');
    expect(fn).not.toHaveBeenCalled();
  });

  it('should work emit change event in group', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckboxGroup, {
      props: {
        modelValue: ['1'],
        onChange: fn
      },
      slots: {

        default: () => h(ECheckbox, { label: '1', value: '1' })
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
    await wrapper.findComponent(ECheckbox).find('input').trigger('change');
    expect(fn).toHaveBeenCalled();
  });

  it('should work with label props', () => {
    const wrapper = mount(ECheckbox, {
      props: {
        label: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.eu-checkbox__label').text()).toBe('test');
  });
});

