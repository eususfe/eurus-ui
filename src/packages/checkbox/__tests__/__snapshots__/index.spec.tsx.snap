// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ECheckbox > ECheckbox snapshot 1`] = `
"<label class=\\"eu-checkbox\\" aria-hidden=\\"true\\"><input type=\\"checkbox\\" class=\\"eu-checkbox__input\\"><span class=\\"eu-checkbox-inner\\"><span class=\\"eu-icon eu-checkbox-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 14px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 64 64\\" class=\\"eu-icon eu-icon-chevron-left\\"><path d=\\"M50.42,16.76L22.34,39.45l-8.1-11.46c-1.12-1.58-3.3-1.96-4.88-0.84c-1.58,1.12-1.95,3.3-0.84,4.88l10.26,14.51  c0.56,0.79,1.42,1.31,2.38,1.45c0.16,0.02,0.32,0.03,0.48,0.03c0.8,0,1.57-0.27,2.2-0.78l30.99-25.03c1.5-1.21,1.74-3.42,0.52-4.92  C54.13,15.78,51.93,15.55,50.42,16.76z\\"></path></svg></span></span>
  <div class=\\"eu-checkbox__label\\">
    <!---->
  </div>
</label>"
`;

exports[`ECheckbox > should work emit change event in group 1`] = `
"<div class=\\"eu-checkbox-group\\"><label class=\\"eu-checkbox eu-checkbox-checked\\" aria-hidden=\\"true\\"><input type=\\"checkbox\\" checked=\\"\\" class=\\"eu-checkbox__input\\" value=\\"1\\"><span class=\\"eu-checkbox-inner\\"><span class=\\"eu-icon eu-checkbox-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 14px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 64 64\\" class=\\"eu-icon eu-icon-chevron-left\\"><path d=\\"M50.42,16.76L22.34,39.45l-8.1-11.46c-1.12-1.58-3.3-1.96-4.88-0.84c-1.58,1.12-1.95,3.3-0.84,4.88l10.26,14.51  c0.56,0.79,1.42,1.31,2.38,1.45c0.16,0.02,0.32,0.03,0.48,0.03c0.8,0,1.57-0.27,2.2-0.78l30.99-25.03c1.5-1.21,1.74-3.42,0.52-4.92  C54.13,15.78,51.93,15.55,50.42,16.76z\\"></path></svg></span></span>
    <div class=\\"eu-checkbox__label\\">1</div>
  </label></div>"
`;

exports[`ECheckbox > should work with label props 1`] = `
"<label class=\\"eu-checkbox\\" aria-hidden=\\"true\\"><input type=\\"checkbox\\" class=\\"eu-checkbox__input\\"><span class=\\"eu-checkbox-inner\\"><span class=\\"eu-icon eu-checkbox-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 14px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 64 64\\" class=\\"eu-icon eu-icon-chevron-left\\"><path d=\\"M50.42,16.76L22.34,39.45l-8.1-11.46c-1.12-1.58-3.3-1.96-4.88-0.84c-1.58,1.12-1.95,3.3-0.84,4.88l10.26,14.51  c0.56,0.79,1.42,1.31,2.38,1.45c0.16,0.02,0.32,0.03,0.48,0.03c0.8,0,1.57-0.27,2.2-0.78l30.99-25.03c1.5-1.21,1.74-3.42,0.52-4.92  C54.13,15.78,51.93,15.55,50.42,16.76z\\"></path></svg></span></span>
  <div class=\\"eu-checkbox__label\\">test</div>
</label>"
`;
